### ISSUE TITLE
addSearchMarker

### ISSUE THREAD

**Issue by Patrikios** [byzheng/leafletplugins#2]
When I try to implement your leafletplugins search button, I bump into the following problem:

the popup is in the form
    popup = ~ as.character(paste0(place.variable, ": ", factor(description.variable))))

In this case 
    addSearchMarker('marker', position='topleft', propertyName =  'popup')

For reproducibility, here the code I would like to run:
    leaflet(cities.custom) %>% 
      setView(lng =A, lat = B, zoom = 8) %>% 
      addProviderTiles("Stamen.Toner") %>% 
      addCircles(group='marker',
                 lng=~lon, 
                 lat=~lat, 
                 radius = ~sqrt(size.variable) * 50, 
                 popup = ~ as.character(paste0(place.variable, ": ", factor(description.variable)))) %>%
      addLegend("bottomright", 
                color= "xyz",
                title = "", 
                labels="xyz") %>%
      addSearchMarker('marker', position='topleft', propertyName =  'popup')

Can the plugin handle the more "complicated" structure of the propertyName like from my example paste0?

**Comment by byzheng** [comment]
Never try the complicated popup. Could you give me some reproducable data (citis.custom)? Then I can try it.

**Comment by Patrikios** [comment]
Even when I try to run your example from the package:

    library(leafletplugins)
    data(quakes)
    leaflet(data = quakes) %>%
      addTiles() %>%
      addMarkers(~long, ~lat, popup = ~as.character(mag),
                 group = 'marker', label = ~as.character(mag)) %>%
      addSearchMarker('marker', position = 'topleft', propertyName = 'popup')

I get an error:

    Error in addMarkers(., ~long, ~lat, popup = ~as.character(mag), group = "marker",  : 
      unused argument (label = ~as.character(mag))

Is the 'label' parameter deprecated? 
When I remove it, still, the search does not work like you nicely illustrated at the github page with the example (https://byzheng.github.io/leafletplugins/search.html). Can you point out the correct direction to make the search plugin work?

**Comment by byzheng** [comment]
Sorry You need to install the develop version of leaflet in github. 


devtools::install_github('byzheng/leaflet')

**Comment by Patrikios** [comment]
yes, that was one thing. I got still the problem though to get searched for addCircles() as in the following min. repr. example:
    data <- structure(list(place = c("ALFHAUSEN", "ANDERVENNE", "ANKUM", 
    "ASCHENDORF", "BAD ESSEN", "BAD IBURG"), number = c(592L,  174L, 1147L, 848L, 2446L,     2010L), 
    group = c("A 103, C 24, B 465", "A 23, C 6, B 145", "A 166, C 65, B 916", "A 133, C 50, B 665", 
    "A 398, C 121, B 1927","A 376, C 137, B 1497"), lon = c(7.95198, 7.571398, 7.871343, 7.334298, 
    8.350827, 8.045282), lat = c(52.50032, 52.50899, 52.54118, 53.05281, 52.31984, 52.15453)), 
    .Names = c("place", "number", "group", "lon", "lat"), row.names = c(NA,  6L), class = "data.frame")                                                                                                                                                                                                                                                                                                                            6L), class = "data.frame")                                                                                                                                                                                                                                                                                                                                            6L), class = "data.frame")

    leaflet(data) %>% 
      setView(lng =7.950778, lat = 52.675599, zoom = 8) %>% 
      addTiles() %>% 
      addCircles(group="marker",
                 lng=~lon, 
                 lat=~lat, 
                 radius = ~sqrt(number) * 50, 
                 label=~place,
                 popup = ~ as.character(paste0(place, ": ", group))) %>%
      addSearchMarker(group="marker", position='topleft', propertyName ='label')

**Comment by byzheng** [comment]
It seems the leaflet-search does not work for circles.

**Comment by Patrikios** [comment]
Do you plan to implement the search button also for AddCircles?

**Comment by mubashirqasim** [comment]
hi there, 
I am struggling to add search feature on SuperZip example of leaflet available from: https://github.com/jcheng5/superzip

I just changed following part in the example:

```
leafletProxy("map", data = zipdata) %>%
      clearShapes() %>%
      addCircleMarkers(~longitude, ~latitude, radius=radius/5000, layerId=~zipcode, label = ~zipcode, group = 'marker',
        stroke=FALSE, fillOpacity=0.4, fillColor=pal(colorData)) %>%
      addLegend("bottomleft", pal=pal, values=colorData, title=colorBy,
        layerId="colorLegend") %>%
      addSearchMarker('marker', position = 'topleft', propertyName = 'label')
  })
```
It runs fine. But when I make a search and a add a search a marker, after that, popup on click stops working. I think search markers add a new layer on the top the the map which stops that.

**Comment by byzheng** [comment]
The search plugin is not working for CircleMarkers
